src/charm.py

```diff
@@ -991,6 +1004,99 @@ def is_hostname(st: Optional[str]) -> bool:
             return None
 
 
+class SecretStore:
+    """Wrapper for storing and retrieving private key data from the backend.
+
+    The backend can either be Juju secrets or stored state,
+    depending on the availability of Juju secrets in the environment.
+    """
+
+    _PRIVATE_KEY_SECRET_LABEL = "PRIVATE_KEY"
+
+    def __init__(self, charm: TraefikIngressCharm):
+        self.charm = charm
+
+    def store_private_key(self, private_key: str) -> None:
+        """Stores the private key in the appropriate backend.
+
+        Args:
+            private_key(str): Private key to store.
+        """
+        if self.using_juju_secrets:
+            content = {"private-key": private_key}
+            self.charm.app.add_secret(content, label=self._PRIVATE_KEY_SECRET_LABEL)
```

Comment:
**Question**: Is it safe to call `add_secret` multiple times with the same content/label?